lib/ReactViews/Custom/Chart/Legends.jsx

```diff
@@ -52,14 +50,40 @@ class Legend extends React.PureComponent {
     const squareSize = 20;
     const Glyph = Glyphs[this.props.glyph] ?? GlyphCircle;
     return (
-      <LegendItem margin="0 5px">
-        <svg width={`${squareSize}px`} height={`${squareSize}px`}>
+      <Label title={label.text}>
+        <svg
+          width={`${squareSize}px`}
+          height={`${squareSize}px`}
+          style={{ flexShrink: 0 }}
+        >
           <Glyph top={10} left={10} fill={label.value} size={100} />
         </svg>
-        <LegendLabel className={Styles.label} align="left" margin="0 0 0 4px">
-          {label.text}
-        </LegendLabel>
-      </LegendItem>
+        <LabelText>{label.text}</LabelText>
+      </Label>
     );
   }
 }
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  margin: auto;
+  padding: 7px;
+  font-size: 0.8em;
+`;
+
+const Label = styled.div`
+  display: flex;
+  flex: 1;
+  align-items: center;
+  justify-content: center;
+  min-width: 0px;
+`;
+
+const LabelText = styled.span`
+  margin-left: 4px;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
```

Comment:
oh i added this to <Text/> as overflowEllipsis

Reply:
I tend to use locally defined components instead of the `Text` `Box` generic ones. I wish we had tailwind.